presto-hive/src/main/java/com/facebook/presto/hive/SimpleBackoff.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.google.common.base.Ticker;
+import io.airlift.units.Duration;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+public class SimpleBackoff
+{
+    private final long minBackoffNanos;
+    private final long maxBackoffNanos;
+    private final int factor;
+    private final Ticker ticker;
+
+    private long currentBackoffNanos;
+    private long lastFailureTimeNanos;
+
+    public SimpleBackoff(Duration minBackoffDuration, Duration maxBackoffDuration, int factor)
+    {
+        this(minBackoffDuration.roundTo(NANOSECONDS), maxBackoffDuration.roundTo(NANOSECONDS), factor, Ticker.systemTicker());
+    }
+
+    public SimpleBackoff(long minBackoffNanos, long maxBackoffNanos, int factor, Ticker ticker)
+    {
+        checkArgument(maxBackoffNanos > minBackoffNanos, "maxBackoff is less than minBackoff");
+        this.maxBackoffNanos = maxBackoffNanos;
+        this.minBackoffNanos = minBackoffNanos;
+        this.ticker = ticker;
+        this.factor = factor;
+    }
+
+    public synchronized void fail()
+    {
+        lastFailureTimeNanos = ticker.read();
+        if (currentBackoffNanos < maxBackoffNanos) {
+            long backoff;
+            try {
+                backoff = Math.multiplyExact(currentBackoffNanos, factor);
+            }
+            catch (ArithmeticException e) {
+                backoff = maxBackoffNanos;
+            }
+            currentBackoffNanos = Math.max(Math.min(backoff, maxBackoffNanos), minBackoffNanos);
```

Comment:
why we need this?
in the formulas above, can;t you directly assign to `currentBackoffNanos`, removing `backoff` var and this line?